pkg/domain/runaway.go

```diff
@@ -308,7 +306,28 @@ func (do *Domain) handleRunawayWatch(record *resourcegroup.QuarantineRecord) err
 	}()
 	sql, params := record.GenInsertionStmt()
 	_, err = exec.ExecuteInternal(ctx, sql, params...)
-	return err
+	if err != nil {
+		return 0, err
+	}
+	for retry := 1; retry <= maxIDRetries; retry++ {
+		var rs sqlexec.RecordSet
+		rs, err = exec.ExecuteInternal(ctx, `SELECT LAST_INSERT_ID();`)
+		if err == nil {
```

Comment:
just a coding style suggestion, we can use 
```
if err != nil {
  continue
}
```

to reduce the indentation. Also for below if-check

Reply:
If we use `continue `, how do I gracefully write the logic to wait for interval between retries?

Reply:
maybe move the sleep to the beginning of loop (skip the sleep for the first try), or use goto 😂